lxd/storage/drivers/driver_powerflex_utils.go

```diff
@@ -864,6 +876,17 @@ func (d *powerflex) mapNVMeVolume(vol Volume) (revert.Hook, error) {
 		})
 	}
 
+	if d.config["powerflex.mode"] == "nvme" {
+		// Connect to the NVMe/TCP subsystem.
+		// We have to connect after the first mapping was established.
+		// PowerFlex does not offer any discovery log entries until a volume gets mapped to the host.
+		// This action is idempotent.
+		err = d.connectNVMeSubsys()
```

Comment:
Does this need a revert to disconnect on error?

Reply:
This was already covered a few lines above with the reverter as part of `unmapVolume`. 
I didn't like that this hides this functionality so I made the following modifications:

1. Let `createNVMeHost` return a reverter if the host actually got created (since it's idempotent)
2. Let `connectNVMeHost` return a reverter if the host isn't already connected to the subsystem (since it's idempotent). This is very important to not disconnect in case other instances are already running in parallel.
3. Remove the `powerflex.host` lock for the creation of NVMe hosts in PowerFlex as the entire process (map/unmap) is already completely locked with the `nvme` lock.

Reply:
With this I fixed a regression introduced by https://github.com/canonical/lxd/pull/13647. It always tried to reconnect to NVMe/TCP subsystem (because the string check was using the wrong value). There was now always a reverter for disconnecting even though the system was already connected. 

It surfaced whilst putting the proper reverters in place :) Thanks for the comment. The commit is https://github.com/canonical/lxd/pull/13660/commits/f05ee3e6ac620e814e663b4cd317aeed173b8c71. 
I have did it in this PR as it's closely related to the reverter changes in here.

Reply:
Well even before https://github.com/canonical/lxd/pull/13647, the check used the wrong value (protection domain id instead of system id). So it's not really a regression, it just continued to use the wrong check.